### PR TITLE
CI: Disable root tests in dev mode

### LINKED ISSUE #29: Don't run unit tests as root in dev mode
`.ci/go-test.sh` should not run tests as `root` if `KATA_DEV_MODE` is set.

### PR DESCRIPTION
**Pull request by jodh-intel**
Don't run the unit tests as root in developer mode (when
`KATA_DEV_MODE`) is set.

Fixes #29.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Hi @chavafg, @sameo, @sboeuf, @grahamwhaley, @bergwolf - what do you think?

If we decide that developers really do want to run tests as `root` locally, my view is that they should be required to set an extra variable (`KATA_DEV_MODE_RUN_AS_ROOT=true` or something).

But it should be considered **very** unsafe for a developer to set `CI` in their environment since all scripts should assume if that variable is set that they are running on a real CI system (*so can do whatever they like*).

**Comment by chavafg** [comment]
I agree that developers should not run the tests as root. 
Could it be possible that a developer does not know about `KATA_DEV_MODE` flag and sets `CI=TRUE`? In this case, will he run tests as `root` without noticing? 
If so, we would need to document how the a developer should run the tests.

**Comment by jodh-intel** [comment]
@chavafg - I agree. We need a `README`. I'm happy to write one (either on this PR or another).

**Comment by jodh-intel** [comment]
Hi @chavafg - I'm going to add a `README` on #31 (as the content in master already mentions `KATA_DEV_MODE`).

**Comment by jodh-intel** [comment]
See #32.

**Comment by chavafg** [comment]
lgtm, thanks for the `README`

**Comment by jodh-intel** [comment]
Could we get one more review on this PR please?

File: .ci/go-test.sh [hunk 1]
```diff
@@ -64,16 +70,22 @@ test_coverage()
...
+		warn "As a result, only a subset of tests will be run"
+		warn "(run this script as a non-privileged to ensure all tests are run)."
 	else
-		# Run the unit-tests *twice* (since some must run as root and
-		# others must run as non-root), combining the resulting test
-		# coverage files.
-		users="current root"
+		if [ -n "$KATA_DEV_MODE" ]; then
+			warn "Dangerous to set CI with KATA_DEV_MODE set"
+			warn "Not also running tests as root"
```
Comment (grahamwhaley) [inline .ci/go-test.sh]: can we rephrase this line - it doesn't quite make sense in my head....

Comment (jodh-intel) [reply .ci/go-test.sh]: Agreed - reworded for improved clarity.

**Pull request merged**